editorconfig.el

```diff
@@ -378,6 +378,32 @@ TRIM-TRAILING-WS."
              (> (string-to-number length) 0))
     (setq fill-column (string-to-number length))))
 
+(defun editorconfig--is-a-mode-p (current want)
+  "Return non-nil if major mode CURRENT is a major mode WANT."
+  (or (eq current
+          want)
+      (let ((parent (get current 'derived-mode-parent)))
+        (and parent
+             (editorconfig--is-a-mode-p parent want)))))
+
+(defun editorconfig-set-major-mode (filetype)
+  "Set buffer `major-mode' by FILETYPE.
+
+FILETYPE should be s string like `\"ini\"`, if not nil or empty string."
+  (let ((mode (and filetype
+                   (not (string= filetype
+                                 ""))
+                   (intern (concat filetype
+                                   "-mode")))))
```

Comment:
While the convention with `name-mode` is very strong, I don't think that every major mode must end in `-mode`. Maybe editorconfig can check to see if the raw name (not ending with `-mode`) is checked as well?

Reply:
On second thought, checking all functions without `-mode` seems to be danger for some cases (for example `file_type_emacs = kill-emacs` !), so I'm going to leave this as it is now...